src/client/clientmap.cpp

```diff
@@ -1323,13 +1323,19 @@ void ClientMap::updateTransparentMeshBuffers()
 
 		if (m_needs_update_transparent_meshes ||
 				blockmesh->getTransparentBuffers().size() == 0) {
+			bool do_sort_block = transparency_sorting_enabled;
+
+			if (do_sort_block) {
+				v3f mesh_sphere_center = intToFloat(block->getPosRelative(), BS)
+						+ blockmesh->getBoundingSphereCenter();
+				f32 mesh_sphere_radius = blockmesh->getBoundingRadius();
+				f32 distance_sq = m_camera_position.getDistanceFromSQ(mesh_sphere_center);
 
-			v3f mesh_sphere_center = intToFloat(block->getPosRelative(), BS)
-					+ blockmesh->getBoundingSphereCenter();
-			f32 mesh_sphere_radius = blockmesh->getBoundingRadius();
-			f32 distance_sq = m_camera_position.getDistanceFromSQ(mesh_sphere_center);
+				if (distance_sq > std::pow(sorting_distance + mesh_sphere_radius, 2.0f))
+					do_sort_block = false;
```

Comment:
if `sorting_distance == 0` will this ever be true anyway?

Reply:
If the `if (do_sort_block) {` was not there, of course, yes. You'd just have to be inside the mesh bounding sphere.